src/constraint.rs

```diff
@@ -17,7 +16,12 @@ pub trait ConstraintValidator {
     /// adding [Violation]s and/or [ViolationChild]ren to `Err(violation)`
     /// if the constraint is violated.
     /// Otherwise, if the value passes the validation against the constraint then returns `Ok(())`.
-    fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult;
+    fn validate(
+        &self,
+        value: &OwnedElement,
+        type_store: &TypeStore,
+        open_content: bool,
```

Comment:
issue—the `content: closed` pseudo-constraint is only ever useful with `fields` constraint, and so adding it here is polluting the API. Some others ways to solve this might be
* Pass the whole type definition to the `validate` function. This is marginally better because it feels less specific, but I think that it's really still just something that the `fields` constraint would use. Constraints shouldn't have to know about other constraints when they are testing a value.
* Add an `open_content: bool` field to `Constraint::Fields` and try to push to problem out to the translation between the `IslConstraint` and the `Constraint`. We might still need to pass around the type definition, but I think it makes more sense here because this is the most logical place (IMO) to resolve other discrepancies (such as this issue or syntactical sugar) between the language and the internal model.

Reply:
Added `open_content` property to `FieldsConstraint` struct. `open_content` boolean value is passed to `FieldsConstraint` while converting from `IslConstraint` to `Constraint`.